### PR TITLE
GH-40566: [C++] Fix 3.12 Python support

### PR DESCRIPTION
**Pull request by SuperSandro2000**
sysconfig is available since Python 3.2

Tested on nixpkgs and works as expected

### Rationale for this change

Support Python 3.12

### Are these changes tested?

Yes

### Are there any user-facing changes?

No

* GitHub Issue: #40566

### CODE REVIEW

File: cpp/cmake_modules/FindPython3Alt.cmake [hunk 1]
```diff
@@ -67,7 +67,7 @@ get_target_property(NUMPY_INCLUDE_DIRS Python3::NumPy INTERFACE_INCLUDE_DIRECTOR
 # detect it ourselves.
 # (https://gitlab.kitware.com/cmake/cmake/issues/20408)
 execute_process(COMMAND "${PYTHON_EXECUTABLE}" "-c"
-                        "from distutils import sysconfig; print(sysconfig.get_config_var('EXT_SUFFIX'))"
+                        "import distutils; print(sysconfig.get_config_var('EXT_SUFFIX'))"
```
Comment (mweinelt) [inline cpp/cmake_modules/FindPython3Alt.cmake:70]:
```suggestion
                        "import sysconfig; print(sysconfig.get_config_var('EXT_SUFFIX'))"
```

**Comment by kou** [comment]
Could you open an issue for this?
See also the auto generated comment: https://github.com/apache/arrow/pull/40322#issuecomment-1974866490

**Comment by SuperSandro2000** [comment]
I currently lack the time and motivation to open an issue for this benial change and then I would rather close this PR and carry the one line patch for the time being.

**Review by assignUser — APPROVED** [review]
Thanks for the fix!

@SuperSandro2000 could you comment on https://github.com/apache/arrow/issues/40566 so I can assign you? (you can also comment 'take' and the bot will do it)

**Pull request merged**